packages/core/src/sfpcommands/source/DeploySourceToOrgImpl.ts

```diff
@@ -216,76 +128,92 @@ export default class DeploySourceToOrgImpl implements DeploymentExecutor {
         SFPLogger.log(table.toString(), LoggerLevel.WARN, this.packageLogger);
     }
 
-    private displayTestFailures(testFailures) {
+    private displayTestFailures(testFailures: Failures | Failures[]) {
         let table = new Table({
             head: ['Test Name', 'Method Name', 'Message'],
         });
 
-        testFailures.forEach((elem) => {
-            table.push([elem.name, elem.methodName, elem.message]);
-        });
-
+        if (Array.isArray(testFailures)) {
+            testFailures.forEach((elem) => {
+                table.push([elem.name, elem.methodName, elem.message]);
+            });
+        } else {
+            table.push([testFailures.name, testFailures.methodName, testFailures.message]);
+        }
         SFPLogger.log('Unable to deploy due to test failures:', LoggerLevel.WARN, this.packageLogger);
         SFPLogger.log(table.toString(), LoggerLevel.WARN, this.packageLogger);
     }
 
-    private buildExecCommand(): string {
-        let apexclasses;
-
-        let command = `sfdx force:mdapi:deploy -u ${this.target_org}`;
+    private async buildDeploymentOptions(sourceDir: string, username: string): Promise<MetadataApiDeployOptions> {
+        username = await convertAliasToUsername(username);
+        let metdataDeployOptions: MetadataApiDeployOptions = {
+            usernameOrConnection: username,
+            apiOptions: {},
+        };
 
-        if (this.deployment_options['checkonly']) command += ` -c`;
-
-        //directory
-        command += ` -d ${this.mdapiDir}`;
-
-        //add json
-        command += ` --json`;
+        if (this.deployment_options['apiversion'])
+            metdataDeployOptions.apiVersion = this.deployment_options['apiversion'];
 
         if (this.deployment_options['testlevel'] == 'RunApexTestSuite') {
-            //testlevel
-            command += ` -l RunSpecifiedTests`;
-            apexclasses = this.convertApexTestSuiteToListOfApexClasses(this.deployment_options['apextestsuite']);
-            command += ` -r ${apexclasses}`;
+            metdataDeployOptions.apiOptions.testLevel = `RunSpecifiedTests`;
+            let apexTestSuite = new ApexTestSuite(sourceDir, this.deployment_options['apextestsuite']);
```

Comment:
Do we pass `this.deployment_options['apextestsuite']` in callers of `DeploySourceToOrgImpl`?

Reply:
Don't think we actually support apex test suite in any of our source deployment commands

Reply:
I think we can keep it for now, there is an incorrect implementation in BuildImpl  where package Metadata has option to pickup apex test suite